### PR TITLE
amqp: Add 'subscribe' method to D-Bus interface

### PR DESCRIPTION
**Pull request by przemyslawfierek-silvair**

### CODE REVIEW

File: mesh/amqp.c [hunk 1]
```diff
@@ -609,6 +631,18 @@ static void control_message_handler(struct amqp_thread_context *context)
 							msg.publish.size);
 			return;
 
+		case SUBSCRIBE:
+			l_queue_push_tail(context->subscriptions,
+					l_strdup(msg.subscribe.pattern));
+
+			if (context->amqp_state == MESH_AMQP_STATE_CONNECTED) {
+				destroy_connection(context);
```
Comment (michallowasrzechonek-silvair) [inline mesh/amqp.c]: This means that calling this api will immediately cause a reconnect... Im not a fan of such violence ;)

File: mesh/amqp.c [hunk 2]
```diff
@@ -610,6 +630,11 @@ static void control_message_handler(struct amqp_thread_context *context)
 							msg.publish.size);
 			return;
 
+		case SUBSCRIBE:
```
Comment (michallowasrzechonek-silvair) [inline mesh/amqp.c:633]: Shouldn't we call amqp_subscribe_topics() after updating the list?

Also, what if topics are duplicated?

Comment (przemyslawfierek-silvair) [reply mesh/amqp.c:633]: Current API requires to call methods in right order and doesn't allow for modifications during run time (Everything must be setup during startup).  Personally I don't like that and I want to change it, but in fact it do everything what we need and currently there is no case to change it.

Comment (michallowasrzechonek-silvair) [reply mesh/amqp.c:633]: Ok, you don't like and I don't really like it, let's fix it then.

Let's assume that area list can change *without* application restart, and let's provide API that works with that assumption, even if it's not true right now.

**Pull request closed without merging**